cparse.nim

```diff
@@ -794,6 +794,8 @@ proc parseStructBody(p: var Parser, stmtList: PNode, isUnion: bool,
     elif p.tok.xkind == pxDirective or p.tok.xkind == pxDirectiveParLe:
       var define = parseDir(p, statement)
       addSon(result, define)
+      if p.tok.xkind == pxCurlyRi:
```

Comment:
I don't understand this fix.

Reply:
Given the code in the PR and the test case in the PR description:
Without the check at line 797 in the proposed fix, after parsing the ifdef block the parser expects the next token to be an identifier (in the *typeAtom* call).  This fails because it's actually the right curly associated with the struct definition.  Furthermore if the code skips the call to *typeAtom* and drops into the subsequent while loop it fails because *baseTyp* is nil. The fix I proposed attempts to bypass those issues and jump to the beginning of the outer while loop where the right curly will be evaluated as part of the struct definition (which it is).

Reply:
Ok.